### DISCUSSION TITLE
Protect hosted documents/pdfs/images in next.js from downloading/copying/printing

### DISCUSSION THREAD

**Discussion by abedshaaban** [vercel/next.js#46247] [Help]
The hosted files in my public folder are not entirely protected; any user can access/ download them without restrictions. Knowing that I don't want to allow users to download my files/ pdfs. they can be accessed by clicking inspect > network  > (specifically Fetch/ XHR) where the URL for the document can be shown in the Header > General > Request URL.

how can I block users from  seeing that and downloading the files/ docs

thx in advanced 😄

**Comment by ciruz** [comment]
With a middleware ofc, where you can make sure the user is logged in or has the correct permissions. Then you can stream the file to the user.

This sounds like a good read: https://hackernoon.com/nextjs-and-file-downloads-three-ways-to-restrict-access-to-a-file

**Reply by abedshaaban** [reply]
i loved the article, however, the files are free for the public to see, thus no sign-in is required at all on the website yet. so I have to search for other approaches to prevent the download thing. + an important thing I discovered from the article is that everything in the public folder can be accessed and search engines can too.

**Reply by Tednoob17** [reply]
You finally solve this issue ?, because actually i have the same problem, i want to block any document downloading fro my page and access only the reading.

**Reply by abedshaaban** [reply]
Nope, didn't find an alternative solution yet.
